test/test_router.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"net/http"
+	"time"
+
+	c "github.com/flynn/go-check"
+)
+
+type RouterSuite struct {
+	Helper
+}
+
+var _ = c.ConcurrentSuite(&RouterSuite{})
+
+func (s *RouterSuite) TestAdditionalHttpPorts(t *c.C) {
+	// boot 1 node cluster
+	x := s.bootCluster(t, 1)
+	defer x.Destroy()
+
+	// update router env
+	t.Assert(x.flynn("/", "-a", "router", "env", "set", "ADDITIONAL_HTTP_PORTS=8080"), Succeeds)
+	time.Sleep(1 * time.Second)
```

Comment:
Is this sleep necessary? `flynn env set` should only return when the router is fully in service, if this is not the case then there is a bug.